cellmlmanip/model.py

```diff
@@ -173,12 +178,19 @@ def connect_variables(self, source_name: str, target_name: str):
 
             logger.debug('Updated target: %s', target)
 
+            # Invalidate cached graphs
+            self._invalidate_cache()
+
             return True
 
         # The source variable has not been assigned a symbol, so we can't make this connection
         logger.info('The source variable has not been assigned to a symbol '
                     '(i.e. expecting a connection): %s ⟶ %s',
                     target.name, source.name)
+
+        # Invalidate cached graphs
+        self._invalidate_cache()
```

Comment:
What you could do is pull out the return statement and invalidate cache as that's double and then return  `source.assigned_to and source.units != target.units. I know it doesn't matter and of course return True/False is easier to write but you'd be less likely to accidentally forget to invalidate cache if things change I think

Reply:
Not sure what you mean here? I think all modification methods should invalidate the cache, as it's too easy to forget otherwise. Unless you mean this method could do so just once, rather than having that line twice?

Reply:
That's what I mean yeah.

Reply:
OK refactoring this method to have nicer flow